module/VuFind/src/VuFind/AjaxHandler/GetFacetData.php

```diff
@@ -109,8 +109,16 @@ public function handleRequest(Params $params)
         } else {
             $facetList = $facets[$facet]['data']['list'];
 
-            if (!empty($sort)) {
-                $this->facetHelper->sortFacetList($facetList, $sort == 'top');
+            switch ($sort) {
```

Comment:
This switch logic is found in two places -- here and in AbstractSolrSearch. Would it make more sense to move it inside the facetHelper so it can be shared more easily? For example, what if we changed the $topLevel parameter to HierarchicalFacetHelper::sortFacetList so that it accepts either a boolean (existing behavior) or a string (this switch behavior)? I think that might make everything a bit more readable and less redundant.

Also, is there a reason that the default sort behavior is 'top' now? The documentation suggests that 'count' is the default behavior.

Reply:
Sort logic is now in HierarchicalFacetHelper::sortFacetList and the default order should be "count" everywhere.